string mode
int16 material
string grip_mode
float32[] coord